src/live_restore/live_restore_state.c

```diff
@@ -0,0 +1,362 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *  All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include "wt_internal.h"
+#include "live_restore_private.h"
+
+/*
+ * __live_restore_state_to_string --
+ *     Convert a live restore state to its string representation.
+ */
+static void
+__live_restore_state_to_string(WTI_LIVE_RESTORE_STATE state, char *state_strp)
+{
+    switch (state) {
+    case WTI_LIVE_RESTORE_STATE_NONE:
+        strcpy(state_strp, "WTI_LIVE_RESTORE_STATE_NONE");
+        break;
+    case WTI_LIVE_RESTORE_STATE_BACKGROUND_MIGRATION:
+        strcpy(state_strp, "WTI_LIVE_RESTORE_STATE_BACKGROUND_MIGRATION");
+        break;
+    case WTI_LIVE_RESTORE_STATE_CLEAN_UP:
+        strcpy(state_strp, "WTI_LIVE_RESTORE_STATE_CLEAN_UP");
+        break;
+    case WTI_LIVE_RESTORE_STATE_COMPLETE:
+        strcpy(state_strp, "WTI_LIVE_RESTORE_STATE_COMPLETE");
+        break;
+    }
+}
+
+/*
+ * __live_restore_state_from_string --
+ *     Convert a string to its live restore state.
+ */
+static int
+__live_restore_state_from_string(
+  WT_SESSION_IMPL *session, char *state_str, WTI_LIVE_RESTORE_STATE *statep)
+{
+    if (strcmp(state_str, "WTI_LIVE_RESTORE_STATE_NONE") == 0)
```

Comment:
Same here - this should use `WT_STRING_LIT_MATCH` for a literal match.

Reply:
Updated in https://github.com/wiredtiger/wiredtiger/pull/11590/commits/2ba6a595f26302ab0d389cef28a2aadc3c202044